public/diagnostic.js

```diff
@@ -23,3 +23,31 @@
 //  -  Order 1 : 2 hats ($5 each)
 //  -  Order 2 : 1 desk lamp ($20 each)
 //  -  Order 3 : 3 hand towels ($8 each)
+import Ember from 'ember';
+
+const Order = Ember.Object.extend({
+  name: 'Default',
+  unitPrice: 0.99,
+  quantity: 1,
+  orderPrice: Ember.computed('unitPrice', 'quantity', function() {
+    return this.get('unitPrice') * this.get('quantity');
+  })
+});
+
+const Cart = Ember.Object.extend({
+  orders: Ember.A([]),
+  addToCart: function (p_name, p_quantity, p_unitPrice) {
```

Comment:
This does work, but it might be easier to assume that the order has already been created.
```js
addToCart: function(order){
    this.get('orders').addObject(order);
  },
```